python/python/bystro/cli/proteomics_cli.py

```diff
@@ -0,0 +1,70 @@
+import argparse
+
+from bystro.api.proteomics import (
+    upload_proteomics_dataset,
+    DatasetTypes
+)
+
+def upload_proteomics_cli(args: argparse.Namespace) -> dict:
+    """
+    Upload a fragpipe-TMT dataset through the /api/jobs/proteomics/ endpoint and 
+    update the annotation job.
+
+    Parameters
+    ----------
+    protein_abundance_file : str
+        Path to the protein abundance file.
+    experiment_annotation_file : str | None
+        Path to the experiment annotation file.
+    experiment_annotation_file : str | None
+        ID of the job associated with the annotation dataset.
+    proteomics_dataset_type : DatasetTypes
+        Type of the proteomics dataset (we only support fragpipe-TMT currently).
+    bystro_credentials_dir : str
+        User directory for authentication state.
+    print_result : bool, optional
+        Whether to print the result of the upload operation, by default True.
+
+    Returns
+    -------
+    dict
+        A json response with annotationID and proteomicsID.
+    """
+    dataset_type_str = args.proteomics_dataset_type.replace("-", "_")
+
+    try:
+        proteomics_dataset_type = DatasetTypes[dataset_type_str]
+    except KeyError:
+        raise ValueError(f"Invalid proteomics dataset type: {args.proteomics_dataset_type}")
+
+    return upload_proteomics_dataset(
+        protein_abundance_file=args.protein_abundance_file,
+        experiment_annotation_file=args.experiment_annotation_file,
+        annotation_job_id=args.annotation_job_id,
+        proteomics_dataset_type=proteomics_dataset_type,
+        print_result=True,
+    )
+
+def add_proteomics_subparser(subparsers) -> None:
```

Comment:
```suggestion

def add_proteomics_subparser(subparsers) -> None:
    proteomics_parser = subparsers.add_parser("proteomics", help="Bystro proteomics dataset operations")

    proteomics_subparser = proteomics_parser.add_subparsers(
        title="commands", dest="command", help="Commands for proteomics datasets"
    )

    # Define a default function for proteomics that shows help if no command is provided
    def proteomics_default_help(_args):
        # Capture the help message
        buffer = io.StringIO()
        proteomics_parser.print_help(file=buffer)
        help_message = buffer.getvalue()

        # Custom message
        custom_message = "Bystro proteomics dataset operations.\n\n"

        # Find the insertion point for the custom message
        insertion_point = help_message.find('options:')
        if insertion_point != -1:
            # Insert the custom message just before "options:"
            help_message = help_message[:insertion_point] + custom_message + help_message[insertion_point:]
        else:
            # If "options:" is not found, prepend the custom message
            help_message = custom_message + help_message

        print(help_message)
        proteomics_parser.exit()

    # Set the default function for the subparser
    proteomics_parser.set_defaults(func=proteomics_default_help)

    # Command 1: upload
    upload_parser = proteomics_subparser.add_parser("upload", help="Upload a proteomics dataset")

    upload_parser.add_argument(
        "--protein_abundance_file", required=True, help="Path to the protein abundance file"
    )
    upload_parser.add_argument(
        "--experiment_annotation_file", help="Path to the experiment annotation file"
    )
    upload_parser.add_argument(
        "--annotation_job_id", help="ID of the annotation job to link with the proteomics dataset"
    )
    upload_parser.add_argument(
        "--proteomics_dataset_type",
        choices=[e.name for e in DatasetTypes],
        default="fragpipe_TMT",
        help="Type of the proteomics dataset",
    )
    upload_parser.set_defaults(func=upload_proteomics_cli)

```

This give the following behavior:

```sh
(bystro) (base) ubuntu@ip-10-98-135-15:~/bystro/python$ bystro-api
Using default credentials path: /home/ubuntu/.bystro/bystro_authentication_token.json
usage: bystro-api [-h] {login,signup,get-user,create-annotation,get-jobs,query,proteomics} ...

Bystro CLI tool for making API calls.

options:
  -h, --help            show this help message and exit

commands:
  {login,signup,get-user,create-annotation,get-jobs,query,proteomics}
    login               Authenticate with the Bystro API
    signup              Sign up to Bystro
    get-user            Handle user operations
    create-annotation   Create an annotation
    get-jobs            Fetch one job or a list of jobs
    query               The OpenSearch query string query, e.g. (cadd: >= 20)
    proteomics          Bystro proteomics dataset operations
```

Calling just the proteomics parser:
```sh
(bystro) (base) ubuntu@ip-10-98-135-15:~/bystro/python$ bystro-api proteomics
usage: bystro-api proteomics [-h] {upload} ...

Bystro proteomics dataset operations.

options:
  -h, --help  show this help message and exit

commands:
  {upload}    Commands for proteomics datasets
    upload    Upload a proteomics dataset
```

as an aside, please remove `print(f"Using default credentials path: {CREDENTIALS_PATH}")` from api/auth.py